src/repositories/jobRepository.ts

```diff
@@ -93,6 +93,15 @@ export class JobRepository extends BaseRepository {
     await this._queueConnector.sendMessage(new JobQueueMessage(jobId, status, 0, taskId), url, delay);
   }
 
+  async findOneAndUpdateExecutionTime(id: string, setValues: { [key: string]: number }): Promise<void> {
+    const query = {
+      _id: new objectId(id),
+    };
+    const update = { $set: setValues };
+    const options = { sort: { priority: -1, createdTime: 1 }, returnNewDocument: true };
+    this.findOneAndUpdate(query, update, options, `Mongo Timeout Error: Timed out while retrieving job`);
```

Comment:
Are we sure we don't need to add a `.catch()` method in case `findOneAndUpdate` throws an error? Or will an error from a promise automatically be ignored?

Reply:
@rayangler good question, from what I understand about Promises with error handling is that if an error occurs it will jump to the nearest catch in our cases it should be handle within this [try/catch](https://github.com/mongodb/docs-worker-pool/blob/85c7b8005894f16d3bf2bfa607394bb9e7b57701/src/repositories/baseRepository.ts#L131).

So nope, it shouldn't be ignored. Instead the handling the error in a catch within the `findOneAndUpdate` method in the `jobRepository` file, it will just use the catch inside the `findOneAndUpdate` inside the `baseRepository` file.

Reply:
It looks like `findOneAndUpdate` catches the error then throws it again. Wouldn't that additional `throw` propagate to a different try/catch block? Just want confirmation that an error here won't cause further errors elsewhere

Reply:
i think this will lead to an `unhandled rejection error`. sorry for the shortsighted suggestion @caesarbell 

thinking of a way we can avoid awaiting but also catch errors? this would require some separate queue/service. think for now,
we can return findOneAndUpdate here

and when we call findOneAndUpdateExecutionTime, we can wrap with 
```
try {
  findOneAndUpdateExecutionTime
} catch(e) {
  logger.save()
  // but continue the code
}
```

Reply:
If we're planning to not `await`, might need to do something like:

```
findOneAndUpdateExecutionTime().catch((e) => {
   logger()
   // but continue the code
})
```

I don't think `try`/`catch` can catch non-awaited errors. 

Or, we can do it within the `findOneAndUpdateExecutionTime` function itself so that the function is always guaranteed to be non-blocking if we ever need to use it in more than one place in the future.

```
this.findOneAndUpdate(query, update, options, `Mongo Timeout Error: Timed out while retrieving job`).catch((e) => {
   logger()
   // but continue the code
})
```

Reply:
@rayangler @seungpark within the `callWithBenchmark` can we just handle the error here? Example like this 

```
try {
      await this._jobRepository.findOneAndUpdateExecutionTime(this.currJob._id, update);
    } catch (error) {
      this._logger.error(this._currJob._id, error.message);
    }
```

Reply:
> If we're planning to not `await`, might need to do something like:
> 
> ```
> findOneAndUpdateExecutionTime().catch((e) => {
>    logger()
>    // but continue the code
> })
> ```
> 
> I don't think `try`/`catch` can catch non-awaited errors.
> 
> Or, we can do it within the `findOneAndUpdateExecutionTime` function itself so that the function is always guaranteed to be non-blocking if we ever need to use it in more than one place in the future.
> 
> ```
> this.findOneAndUpdate(query, update, options, `Mongo Timeout Error: Timed out while retrieving job`).catch((e) => {
>    logger()
>    // but continue the code
> })
> ```

catching and returning within `findOneAndUpdateExecutionTime` sounds ideal. it can be non blocking for rest of logic besides logging the times 👍

Reply:
We will have the following 

```
this._jobRepository.findOneAndUpdateExecutionTime(
      query,
      update,
      options,
      `Mongo Timeout Error: Timed out while updating job with timestamps for ${this.currJob._id}`
    ).catch(error => {
      this._logger.save(this.currJob._id, error.message)
    })
``` 
and 

```
async findOneAndUpdateExecutionTime<Query, UpdateValues, Options>(query: Query, update: UpdateValues, options: Options, errorMsg: string): Promise<void> {
    this.findOneAndUpdate(query, update, options, errorMsg);
  }
```